### PR TITLE
Document stock validation of variant overrides

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

Follow up from https://github.com/openfoodfoundation/openfoodnetwork/pull/3505#pullrequestreview-205530320

Our new stock validation logic can be a bit confusing. I added a table for a better overview to understand the different states.

#### What should we test?

No testing required.

#### Release notes

Added: Added code documentation of stock validation rules

Changelog Category: Added

### CODE REVIEW

File: app/models/concerns/stock_settings_override_validation.rb [hunk 1]
```diff
@@ -1,3 +1,29 @@
...
+#
+# - true
+# - false
+# - nil
+#
+# `count_on_hand` can either be:
+#
+#  - nil
+#  - a number
+#
```
Comment (luisramos0) [inline app/models/concerns/stock_settings_override_validation.rb]: My preference goes for using less lines in comments so I'd do:
```suggestion
# `on_demand` can have three values: true, false or nil
# `count_on_hand` can either be: nil or a number 
```

**Comment by luisramos0** [comment]
@kristinalim I think with this comment I prefer not to change this on_demand.nil? to user_producer_stock_settings? but I can change it if you prefer. let me know and I make the change.

**Comment by kristinalim** [comment]
Sounds good to me, @luisramos0. :slightly_smiling_face: Happy to approve #3505 once green.

File: app/models/concerns/stock_settings_override_validation.rb (hunk 1 above)
Comment (mkllnk) [reply app/models/concerns/stock_settings_override_validation.rb]: Good idea. I committed your changes.

**Comment by mkllnk** [comment]
@kristinalim It looks like @luisramos0 is not introducing the `use_producer_stock_settings?` method anymore. I still think that it aids the understanding to have it in the table because it is used in the user interface. In this context `use_producer_settings?` makes sense. Do you think we need to denote that it's not a method? Should we replace `_` with spaces?

**Comment by kristinalim** [comment]
@mkllnk Actually, #3505 is already using `use_producer_stock_settings?`. The PR is already in Test Ready, so the method name in the comment here can be renamed?

**Comment by mkllnk** [comment]
I'll wait for the other PR being merged.

**Comment by mkllnk** [comment]
Okay, @kristinalim I updated the documentation now.

**Pull request merged**